### PR TITLE
Add support for named pipes and sockets

### PR DESCRIPTION
**Pull request by nehaljwani**
This commits adds asserts for named pipe (FIFO) and socket file types.

### CODE REVIEW

File: testpath/asserts.py [hunk 1]
```diff
@@ -137,3 +139,55 @@ def assert_not_islink(path, msg=None):
...
+    """
+    path = _strpath(path)
+    st = _stat_for_assert(path, follow_symlinks, msg)
+    if stat.S_ISFIFO(st.st_mode):
+        if msg is None:
+            msg = "Path is a named pipe: %r" % path
+        raise AssertionError(msg)
+
+def assert_issocket(path, follow_symlinks=True, msg=None):
+    """Assert that path exists and is a named pipe (FIFO).
```
Comment (takluyver) [inline testpath/asserts.py]: These docstrings should probably say 'Unix domain socket' rather than 'named pipe'.

File: testpath/asserts.py [hunk 2]
```diff
@@ -137,3 +139,55 @@ def assert_not_islink(path, msg=None):
         if msg is None:
             msg = "Path is a symlink: %r" % path
         raise AssertionError(msg)
+
+def assert_ispipe(path, follow_symlinks=True, msg=None):
+    """Assert that path exists and is a named pipe (FIFO).
+
+    With follow_symlinks=True, the default, this will pass if path is a symlink
+    to a regular file. With follow_symlinks=False, it will fail in that case.
```
Comment (takluyver) [inline testpath/asserts.py]: 'regular file' -> 'named pipe' 

Similarly in all four of the new docstrings.

Comment (nehaljwani) [inline testpath/asserts.py]: Fixed

File: testpath/asserts.py (hunk 1 above)
Comment (nehaljwani) [inline testpath/asserts.py]: Fixed

**Comment by takluyver** [comment]
Thanks!

**Comment by takluyver** [comment]
I've added them to the docs as well:
http://testpath.readthedocs.io/en/latest/fsasserts.html#unix-specific

**Comment by nehaljwani** [comment]
Ah! A new release already! Cool :+1:

**Comment by takluyver** [comment]
I haven't actually done a release yet - I'll give it a little while in case there's other things to add soon. But I'm pre-emptively marking them as 'new in' to give people reading the docs a hint why those functions aren't there for them.

**Pull request merged**